os/fs/smartfs/smartfs_smart.c

```diff
@@ -893,6 +893,15 @@ static ssize_t smartfs_write(FAR struct file *filep, const char *buffer, size_t
 #endif
 
 			ret = FS_IOCTL(fs, BIOC_WRITESECT, (unsigned long)&readwrite);
+
```

Comment:
No, write restores only sector, it will check sync if last operation is T_WRITE.
You should not finish here.